pkg/aws/ec2/ec2.go

```diff
@@ -195,8 +195,8 @@ func parseENI(iface *ec2.NetworkInterface, vpcs ipamTypes.VirtualNetworkMap, sub
 	}
 
 	for _, ip := range iface.PrivateIpAddresses {
-		if ip.PrivateIpAddress != nil {
-			eni.Addresses = append(eni.Addresses, *ip.PrivateIpAddress)
+		if ip.PrivateIpAddress != nil && !ip.Primary {
+			eni.Addresses = append(eni.Addresses, aws.ToString(ip.PrivateIpAddress))
```

Comment:
@gandro do these line need changing or okay as is for 1.9?

Reply:
As discussed offline:
```suggestion
		if ip.PrivateIpAddress != nil && (ip.Primary == nil || !*ip.Primary) {
			eni.Addresses = append(eni.Addresses, *ip.PrivateIpAddress)
```